### PR TITLE
Improvements to search view in admin

### PR DESCRIPTION
**Pull request by kaedroho**
This PR adds the following features to the search admin view:
 
 - Search by content type (using the new facet feature introduced by https://github.com/wagtail/wagtail/pull/4508)
 - Order by title, updated at or status

screenshot-2018-5-3 wagtail - search

I'll need some help with making it look pretty.

### CODE REVIEW

**Comment by SalahAdDin** [comment]
Great man! I just was thinking about if `Wagtail` provides facets as `Wagtail` uses `ElasticSarch` and one of the most useful features from `django-haystack` was indeed `facets`. 

Will be `facets` enabled for use in templates?

Thank you.

**Comment by kaedroho** [comment]
Yep, facets is being added as an API you can use in your own search template. See https://github.com/wagtail/wagtail/pull/4508

File: wagtail/admin/templates/wagtailadmin/pages/search_results.html [hunk 1]
```diff
@@ -11,14 +11,35 @@ <h2>
 
         {% search_other %}
 
-        {% include "wagtailadmin/pages/listing/_list_explore.html" with show_parent=1 %}
+        {% if pages.object_list.supports_facet %}
+            <nav class="listing-filter">
+                <h3 class="filter-title">{% trans "Content types" %}</h3>
```
Comment (gasman) [inline wagtail/admin/templates/wagtailadmin/pages/search_results.html]: Unless anyone objects, I'll change this to "Page types" as I think that's a more meaningful term for editors.

**Comment by gasman** [comment]
Merged in bf5d6e3363130ee7c2676e3df34f3a57369fab10 + parents. Thanks @kaedroho!

File: wagtail/admin/templates/wagtailadmin/pages/search_results.html [hunk 2]
```diff
@@ -11,14 +11,35 @@ <h2>
...
+                <ul class="filter-options">
+                    {% if not selected_content_type %}
+                        <li style="background-color: #E6E6E6">All ({{ all_pages.count }})</li>
+                    {% else %}
+                        <li><a href="{% url 'wagtailadmin_pages:search' %}?q={{ query_string|urlencode }}">All ({{ all_pages.count }})</a></li>
+                    {% endif %}
```
Comment (BertrandBordage) [inline wagtail/admin/templates/wagtailadmin/pages/search_results.html]: Oh, too bad I didn’t see it: `All (…)` is not marked for translation :\

**Pull request closed without merging**